src/FSharpPlus/Extensions/Seq.fs

```diff
@@ -168,10 +168,12 @@ module Seq =
     ///
     /// <returns>The result sequence.</returns>
     let drop count (source: seq<_>) =
-        let mutable i = count
-        use e = source.GetEnumerator ()
-        while (i > 0 && e.MoveNext ()) do i <- i-1
-        seq { while e.MoveNext () do yield e.Current }
+        seq {
```

Comment:
If this implementation is less efficient/desirable for some reason (although I don't believe it's creating any additional overhead and it's still lazy) then another solution that is closer to the old code is

```fsharp
let drop count (source: seq<_>) =
    seq {
        let mutable i = count
        use e = source.GetEnumerator ()
        while (i > 0 && e.MoveNext ()) do i <- i-1
        while e.MoveNext () do yield e.Current 
    }
```

As by moving the `seq` around the `use` statement on `GetEnumerator()` it means that it won't be disposed until the `seq` computation has completed.

Reply:
Yes, I think that was the original intention in the existing code.